radis/lbl/factory.py

```diff
@@ -1084,8 +1085,13 @@ def eq_spectrum_gpu(
 
         # load the data
         df = self.df0
-        iso = df["iso"].to_numpy(dtype=np.uint8)
         v0 = df["wav"].to_numpy(dtype=np.float32)
+
+        if len(iso_set) > 1:
+            iso = df["iso"].to_numpy(dtype=np.uint8)
+        elif len(iso_set) == 1:
+            iso = np.full(len(v0), iso_set[0], dtype=np.uint8)
```

Comment:
The GPU code is not so easy to customize so it's quite a bit of extra code for only a small gain to do this (it's stored as a single uint8 which is only 1 byte long) but it's something to consider if we're really pressed for VRAM.

Reply:
1 uint8 * 30 million lines is a lot, so we'll have to get there at some point

Reply:
My point was there are 7 floats for every line and 1 uchar (=iso) so it's a 3.4% save regardless of number of lines. I think there is lower hanging fruit than this we should focus on (i.e. using fixed point types instead of floats)

Nevertheless, I thought about it a bit more and it should actually be easy to implement, will submit a PR for this.